src/storage/storage.cpp

```diff
@@ -557,151 +553,146 @@ void Storage::PopulateData(const DataLayout &layout, char *memory_ptr)
+
+        auto geometries_rev_duration_list_ptr = layout.GetBlockPtr<EdgeWeight, true>(
+            memory_ptr, storage::DataLayout::GEOMETRIES_REV_DURATION_LIST);
+        util::vector_view<EdgeWeight> geometry_rev_duration_list(
+            geometries_rev_duration_list_ptr,
+            layout.num_entries[storage::DataLayout::GEOMETRIES_REV_DURATION_LIST]);
+
+        auto datasources_list_ptr = layout.GetBlockPtr<DatasourceID, true>(
+            memory_ptr, storage::DataLayout::DATASOURCES_LIST);
+        util::vector_view<DatasourceID> datasources_list(
+            datasources_list_ptr, layout.num_entries[storage::DataLayout::DATASOURCES_LIST]);
+
+        extractor::SegmentDataView segment_data{std::move(geometry_begin_indices),
+                                                std::move(geometry_node_list),
+                                                std::move(geometry_fwd_weight_list),
+                                                std::move(geometry_rev_weight_list),
+                                                std::move(geometry_fwd_duration_list),
+                                                std::move(geometry_rev_duration_list),
+                                                std::move(datasources_list)};
+
+        extractor::files::readSegmentData(config.geometries_path, segment_data);
     }
 
     {
         const auto datasources_names_ptr = layout.GetBlockPtr<extractor::Datasources, true>(
             memory_ptr, DataLayout::DATASOURCES_NAMES);
-        extractor::io::read(config.datasource_names_path, *datasources_names_ptr);
+        extractor::files::readDatasources(config.datasource_names_path, *datasources_names_ptr);
     }
 
     // Loading list of coordinates
     {
-        io::FileReader nodes_file(config.nodes_data_path, io::FileReader::HasNoFingerprint);
-        nodes_file.Skip<std::uint64_t>(1); // node_count
         const auto coordinates_ptr =
             layout.GetBlockPtr<util::Coordinate, true>(memory_ptr, DataLayout::COORDINATE_LIST);
         const auto osmnodeid_ptr =
             layout.GetBlockPtr<std::uint64_t, true>(memory_ptr, DataLayout::OSM_NODE_ID_LIST);
-        util::PackedVector<OSMNodeID, true> osmnodeid_list;
-
-        osmnodeid_list.reset(osmnodeid_ptr, layout.num_entries[DataLayout::OSM_NODE_ID_LIST]);
+        util::vector_view<util::Coordinate> coordinates(
+            coordinates_ptr, layout.num_entries[DataLayout::COORDINATE_LIST]);
+        util::PackedVectorView<OSMNodeID> osm_node_ids;
+        osm_node_ids.reset(osmnodeid_ptr, layout.num_entries[DataLayout::OSM_NODE_ID_LIST]);
 
-        serialization::readNodes(nodes_file,
-                                 coordinates_ptr,
-                                 osmnodeid_list,
-                                 layout.num_entries[DataLayout::COORDINATE_LIST]);
+        extractor::files::readNodes<true>(config.nodes_data_path, coordinates, osm_node_ids);
```

Comment:
`true` can be derived from arguments, but better to use template arguments for `coordinates` and `osm_node_ids`